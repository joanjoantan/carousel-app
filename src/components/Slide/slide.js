import React from 'react'

const Slide = ({item, isShowing}) => {
  return <li className="item-container">
    <div className={`item ${isShowing? 'showing': ''}`}>
      <img src={item.previewURL} alt={item.tags}/>
      <div className="itemName">{item.tags}</div>
    </div>
  </li>
}

export default Slide
