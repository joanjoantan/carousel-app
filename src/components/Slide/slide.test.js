import React from 'react'
import { shallow } from 'enzyme'

import Slide from './slide'

describe('Slide component', () => {
  it('should render component with mock value', () => {
    const slide = [{'previewURL': 'url', 'tags': 'name'}]
    const component = shallow(<Slide
      item={slide}
      isShowing='1'/>)
    expect(component).toMatchSnapshot()
  })
})