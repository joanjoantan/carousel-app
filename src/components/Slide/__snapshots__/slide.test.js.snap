// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slide component should render component with mock value 1`] = `
<li
  className="item-container"
>
  <div
    className="item showing"
  >
    <img />
    <div
      className="itemName"
    />
  </div>
</li>
`;
